io/input/chunk_inputformat.cpp

```diff
@@ -45,20 +79,24 @@ bool ChunkInputFormat::next(boost::string_ref& ref) {
     if (buffer_.size() == 0) {
         // fetch new block
         bool success = fetch_new_block();
-        if (!success)
+        if (!success) {
+            is_setup_ = ChunkInputFormatSetUp::NotSetUp;
             return false;
+        }
     }
     if (r == buffer_.size()) {
         // last character in block
         bool success = fetch_new_block();
-        if (!success)
+        if (!success) {
+            is_setup_ = ChunkInputFormatSetUp::NotSetUp;

```

Comment:
This is to check whether the url is set or not. And if it is not success, it has returned false already. Consider such a case, user can set two urls but not doing the reading:
```
infmt.set_input(a);
infmt.set_input(b);
```
So in the end it will use url b instead of a.